scalalib/src/mill/scalalib/TestModuleUtil.scala

```diff
@@ -173,15 +209,217 @@ private[scalalib] object TestModuleUtil {
         else Result.Success((rights.map(_._1).mkString("\n"), rights.flatMap(_._2)))
     }
 
-    subprocessResult match {
-      case Result.Failure(errMsg) => Result.Failure(errMsg)
-      case Result.Success((doneMsg, results)) =>
-        if (results.isEmpty && selectors.nonEmpty) throw doesNotMatchError
-        try handleResults(doneMsg, results, Task.ctx(), testReportXml)
-        catch {
-          case e: Throwable => Result.Failure("Test reporting failed: " + e)
+    subprocessResult
+  }
+
+  private def runTestStealingScheduler(
+      filteredClassLists: Seq[Seq[String]]
+  )(implicit ctx: mill.api.Ctx) = {
+
+    type StealLogCallback = String => Unit
+    val stealLogMap =
+      new java.util.concurrent.ConcurrentHashMap[(Int, Int), (os.Path, StealLogCallback)]()
+
+    def prepareTestSelectorFolder(selectors2: Seq[String], base: os.Path): os.Path = {
+      val selectorFolder = base / "selectors"
```

Comment:
Maybe call this `test-classes` since we know they all refer to concrete class names